Data_Engine/Query/Linq/FilterBy.cs

```diff
@@ -0,0 +1,214 @@
+                if (ignoreStringCase && exactStringMatch)
+                {
+                    if (!isNested)
+                        return objects.Where(x => string.Equals((string)prop.GetValue(x), stringValue, System.StringComparison.CurrentCultureIgnoreCase)).ToList();
+                    else
+                        return objects.Where(x => string.Equals((string)BHRE.Query.PropertyValue(x, propertyName), stringValue, System.StringComparison.CurrentCultureIgnoreCase)).ToList();
+                }
+                if (!exactStringMatch)
+                {
+                    if (!isNested)
+                    {
+                        if (ignoreStringCase)
+                            return objects.Where(x => (prop.GetValue(x) as string).ToUpper().Contains(stringValue.ToUpper())).ToList();
+                        else
+                            return objects.Where(x => (prop.GetValue(x) as string).Contains(stringValue)).ToList();
+                    }
+                    else
+                    {
+                        if (ignoreStringCase)
+                            return objects.Where(x => (BHRE.Query.PropertyValue(x, propertyName) as string).ToUpper().Contains(stringValue.ToUpper())).ToList();
+                        else
+                            return objects.Where(x => (BHRE.Query.PropertyValue(x, propertyName) as string).Contains(stringValue)).ToList();
+                    }
+                }
+                if (!isNested)
+                    return objects.Where(x => prop.GetValue(x).Equals(stringValue)).ToList();
+                else
+                    return objects.Where(x => BHRE.Query.PropertyValue(x, propertyName).Equals(stringValue)).ToList();
+            }
+
+            if (type == typeof(System.Double))
+            {
+                double doubleValue = 0.0;
+                string searchValue = value.ToString();
+                int rounding = searchValue.Contains('.') ? searchValue.Split('.').Last().Length : 0;
+
+                try
+                { 
+                    double.TryParse(searchValue, out doubleValue); 
+                }
+                catch
+                { 
+                    BHRE.Compute.RecordError("Property value to search for cannot be converted to a number");
+                }
+
+                if (!isNested && !isCustomData)
+                    return objects.Where(x => System.Math.Round((double)prop.GetValue(x), rounding) == doubleValue).ToList();
+                else
+                    return objects.Where(x => System.Math.Round((double)BHRE.Query.PropertyValue(x, propertyName), rounding) == doubleValue).ToList();
+            }
+
+            if (prop != null && !isNested)
+            {
+                dynamic val = System.Convert.ChangeType(value, prop.PropertyType);
+                return objects.Where(x => (System.Convert.ChangeType(prop.GetValue(x), type) as dynamic) == val).ToList();
```

Comment:
Not sure I am clear why you need to do this.
What about this?
```c#
if (value != null) 
    return objects.Where(x => value.Equals(prop.GetValue(x))).ToList();
else 
    return objects.Where(x =>prop.GetValue(x) == null).ToList();
```

Reply:
This was me early on and needing an equality for different cases (strings and objects from memory?) as one did not cover the other.